\abstract{\textbf{Motivation:} Breast cancer can be divided into at least three distinct molecular subgroups for which prognosis and treatment outcomes differ significantly.  Accurate subgroup diagnosis is therefore a prerequisite of effective breast cancer treatment.  Previous research has shown that chromosomal copy number aberrations (CNAs) vary between molecular subgroups.  Here we build and evaluate the performance of machine learning classification models using CNA data from 100 breast tumours labelled as one of three subgroups: HER2+, HR+ and Triple Negative.  We evaluate two feature selection methods and three classification methods.\\
\textbf{Results:} We find a combination of Boruta for feature selection and a neural network classifier to be the best performing model with an overall accuracy of 0.895, and find 38 genomic regions to be important for classification between the three subgroups.  In these regions we find several genes that have been found to be involved in the formation of tumours of specific subgroups and identify the HER2+ amplicon that is specific to HER2+ tumour formation and proliferation.  The use of accurate machine learning classifiers to predict breast cancer subgroups from CNA data may lead to faster and more accurate diagnosis and better treatment outcomes.\\
\textbf{Availability:} Scripts and data are available at \href{https://github.com/krademaker/CATS}{https://github.com/krademaker/CATS}}
